### PR TITLE
#161967082 Setup test environment 

### PR DESCRIPTION
**Pull request by Darius-Ndubi**
#### What does this PR do?
- Setup the testing environment

#### Description of Task to be completed?
- Setup unit test environment.
- Setup end to end test environment

#### What are the relevant pivotal tracker stories?
[#161967082](https://www.pivotaltracker.com/story/show/161967082)

#### How should this be manually tested?
- Clone this repo from the terminal
```git clone https://github.com/andela/ah-code-titans-frontend.git```

- Then cd into the projects folder and checkout to the specific branch
```
cd ah-code-titans-frontend
git checkout ch-setup-test-environment-161967082
```
- Install all dependencies
```npm install```
- Test the app
```npm test```

#### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

#### Screenshots
screenshot 2019-01-08 at 16 20 17

### CODE REVIEW

**Review by bettblake08 — APPROVED** [review]
👍

**Review by jillo-abdullahi — APPROVED** [review]
👍

File: .eslintrc [hunk 1]
```diff
@@ -4,8 +4,18 @@
     "browser": true,
     "jest": true
   },
+  "parserOptions": {
+    "ecmaVersion": 7,
```
Comment (justmesam) [inline .eslintrc]: use `"ecmaVersion": 6,`

**Review by justmesam — CHANGES_REQUESTED** [review]
Look at the comments

File: .eslintrc (hunk 1 above)
Comment (Darius-Ndubi) [reply .eslintrc]: Let me make that change,  thank you.

Comment (Darius-Ndubi) [reply .eslintrc]: @justmesam  I have implemented the changes.

File: .eslintrc [hunk 2]
```diff
@@ -4,8 +4,18 @@
...
   },
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true,
+      "modules": true
+    }
+  },
   "extends": "airbnb",
```
Comment (justmesam) [inline .eslintrc:16]: you can add this plugins,
`  "plugins": [
      "react",
      "jsx-a11y",
      "import"
  ]`

Comment (Darius-Ndubi) [reply .eslintrc:16]: @justmesam  I have implemented the changes. Thank you.

**Review by denomash — APPROVED** [review]
👍

**Review by Bryan-Cee — APPROVED** [review]

File:  .hound.yml [hunk 3]
```diff
@@ -1,3 +1,9 @@
+jshint:
```
Comment (justmesam) [inline  .hound.yml]: Remove this file cause is useless now.

**Review by justmesam — CHANGES_REQUESTED** [review]

File:  .hound.yml (hunk 3 above)
Comment (Darius-Ndubi) [reply  .hound.yml]: @justmesam, I have implemented the changes. Thank you.

**Review by justmesam — APPROVED** [review]

**Pull request merged**